plugins/out_opensearch/opensearch.c

```diff
@@ -902,9 +906,26 @@ static void cb_opensearch_flush(struct flb_event_chunk *event_chunk,
     pack = (char *) out_buf;
     pack_size = out_size;
 
+    /* Should we compress the payload ? */
+    if (ctx->compress_gzip == FLB_TRUE) {
+        ret = flb_gzip_compress((void *) pack, pack_size,
+                                &final_payload_buf, &final_payload_size);
+        if (ret == -1) {
+            flb_plg_error(ctx->ins,
```

Comment:
If the code follows this path then by the end of the conditional `final_payload_buf` will be `NULL` and `final_payload_size` will be `0`.

A better approach would be to move the default sets you are doing in 922 to 908 so they are unconditionaly done and then overwrite them in the conditional branch in 918 after setting `compressed` to `FLB_TRUE`.

Reply:
True, good catch! Attemted to fix it in [043f92e](https://github.com/fluent/fluent-bit/pull/7381/commits/043f92e3960f8ec45ca8e94394a0163431c6cc22).